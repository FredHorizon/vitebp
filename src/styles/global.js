import { createGlobalStyle } from 'styled-components';

export const GlobalStyles = createGlobalStyle`
  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }

  html {
    @media(max-width: 1080px) {
      font-size: 93.75%;
    }
    @media (max-width: 720px) {
      font-size: 87.5%;
    }
  }

  body {
    background: ${({ theme }) => theme.colors.secondary};
    -webkit-font-smoothing: antialiased;
  }

  body, input, textarea, button {
    font-family: 'Poppins', sans-serif;
    font-weight: 400;
  }

  h1, h2, h3, h4, h5, strong {
    font-weight: 600;
  }

  button {
    cursor: pointer;
  }
`;
