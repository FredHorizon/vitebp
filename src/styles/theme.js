export const theme = {
  colors: {
    primary: 'purple',
    secondary: 'dimgray',
  },
};
