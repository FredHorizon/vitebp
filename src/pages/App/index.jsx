import { Container } from './styles';
import P from 'prop-types';

function App({ children }) {
  return (
    <Container>
      <div>
        <h1>{children}</h1>
      </div>
    </Container>
  );
}

export default App;

App.propTypes = {
  children: P.node.isRequired,
};
