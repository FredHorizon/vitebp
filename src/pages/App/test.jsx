import { screen } from '@testing-library/react';
import App from './index';
import { renderTheme } from '../../styles/render-theme';

it('should test', () => {
  renderTheme(<App>Test</App>);
  expect(screen.getAllByRole('heading', { name: 'Test' })).toMatchSnapshot();
});
