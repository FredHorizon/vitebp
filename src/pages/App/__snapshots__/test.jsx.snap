// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test 1`] = `
Array [
  .c0 {
  background: purple;
}

<h1
    class="c0"
  >
    <div>
      <h1>
        Test
      </h1>
    </div>
  </h1>,
  <h1>
    Test
  </h1>,
]
`;
